.github/workflows/linux-build-and-test.yaml

```diff
@@ -0,0 +1,144 @@
+            type: Release
+    env:
+      PG_SRC_DIR: /home/runner/postgresql
+    steps:
+
+    - name: Install Dependencies
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      run: sudo apt install lcov
+
+    - name: Cache PostgreSQL
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      id: cache-postgresql
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.PG_SRC_DIR }}
+        key: ${{ runner.os }}-postgresql-${{ matrix.pg }}-${{ matrix.build_type }}
+
+    - name: Build PostgreSQL ${{ matrix.pg }}
+      if: ( matrix.on_pullrequest || github.event_name != 'pull_request' ) && steps.cache-postgresql.outputs.cache-hit != 'true'
+      run: |
+        wget -q -O postgresql.tar.bz2 https://ftp.postgresql.org/pub/source/v${{ matrix.pg }}/postgresql-${{ matrix.pg }}.tar.bz2
+        mkdir -p $PG_SRC_DIR
+        tar --extract --file postgresql.tar.bz2 --directory $PG_SRC_DIR --strip-components 1
+        cd $PG_SRC_DIR
+        ./configure --prefix=/usr/local ${{ matrix.pg_build_args }} --with-openssl --with-llvm --without-readline --without-zlib
+        make
+        make -C src/test/isolation
+        make -C contrib/postgres_fdw
+
+    - name: Install PostgreSQL ${{ matrix.pg }}
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      run: |
+        sudo make -C $PG_SRC_DIR install
+        sudo make -C $PG_SRC_DIR/contrib/postgres_fdw install
+
+    - name: Checkout TimescaleDB
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      uses: actions/checkout@v2
+
+    - name: Test telemetry without OpenSSL
+      if: github.event_name != 'pull_request' && matrix.build_type == 'Debug'
+      run: |
+        BUILD_DIR=nossl ./bootstrap -DCMAKE_BUILD_TYPE=Debug -DPG_SOURCE_DIR=$PG_SRC_DIR -DUSE_OPENSSL=false
+        make -C nossl
+        sudo make -C nossl install
+        make -C nossl installcheck TESTS=telemetry
+
+    - name: Build TimescaleDB
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      run: |
+        ./bootstrap -DCMAKE_BUILD_TYPE="${{ matrix.build_type }}" -DPG_SOURCE_DIR=$PG_SRC_DIR ${{ matrix.tsdb_build_args }}
+        make -C build
+        sudo make -C build install
+
+    - name: Check exported symbols
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      run: ./build/scripts/export_prefix_check.sh
+
+    - name: make installcheck
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      id: installcheck
+      run: make -k -C build installcheck
+
+    - name: coverage
+      if: (matrix.on_pullrequest || github.event_name != 'pull_request') && matrix.coverage
+      run: make -k -C build coverage
+
+    - name: Upload coverage report
+      if: (matrix.on_pullrequest || github.event_name != 'pull_request') && matrix.coverage
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./build/codecov/timescaledb-codecov.info
```

Comment:
While I don't mind saving this file, I am wondering if we are dropping usage of codecov.io?

FWIW, if we generate the HTML output, which now works, then we could upload it to, e.g., s3 bucket or similar for viewing.

Reply:
this step uploads to codecov.io